### PR TITLE
Fix black-box attribute attack to support 1-hot-encoded and scaled features

### PR DESCRIPTION
**Pull request by abigailgold**
Signed-off-by: abigailt <abigailt@il.ibm.com>

# Description

In addition to the ability to attack 1-hot encoded features, this adds support for 1-hot encoded features that have also undergone scaling (so are represented by arbitrary floating point values instead of 0 and 1).

## Type of change

Please check all relevant options.

- [X] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

All unit tests pass

**Test Configuration**:
- OS: MacOS 10.15.7
- Python version: 3.7
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/utils.py [hunk 1]
```diff
@@ -468,6 +468,26 @@ def float_to_categorical(labels, nb_classes=None):
     return categorical
 
 
+def floats_to_one_hot(labels):
```
Comment (beat-buesser) [inline art/utils.py:471]: Could you please add typing and remove the `:type ... ` lines from the docstring?

File: tests/attacks/inference/attribute_inference/test_black_box.py [hunk 2]
```diff
@@ -31,7 +31,7 @@
 from art.estimators.classification import ClassifierMixin
 from art.estimators.classification.scikitlearn import ScikitlearnDecisionTreeClassifier
 
-from tests.attacks.utils import backend_test_classifier_type_check_fail
+# from tests.attacks.utils import backend_test_classifier_type_check_fail
```
Comment (beat-buesser) [inline tests/attacks/inference/attribute_inference/test_black_box.py:34]: Is this change intended?

File: tests/attacks/inference/attribute_inference/test_black_box.py [hunk 3]
```diff
@@ -229,5 +229,5 @@ def test_errors(art_warning, tabular_dl_estimator_for_attack, get_iris_dataset):
         art_warning(e)
 
 
-def test_classifier_type_check_fail():
-    backend_test_classifier_type_check_fail(AttributeInferenceBlackBox, (BaseEstimator, ClassifierMixin))
+# def test_classifier_type_check_fail():
+#     backend_test_classifier_type_check_fail(AttributeInferenceBlackBox, (BaseEstimator, ClassifierMixin))
```
Comment (beat-buesser) [inline tests/attacks/inference/attribute_inference/test_black_box.py:233]: Is this change intended?

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @abigailgold Thank you for the pull request. The code looks good. I have a few questions about some of the changes. Would it be possible to add a unit tests? The test could also visualise how the new method is intended to be used.

File: tests/attacks/inference/attribute_inference/test_black_box.py (hunk 3 above)
Comment (abigailgold) [reply tests/attacks/inference/attribute_inference/test_black_box.py]: No, sorry about that. I committed the wrong file. Undoing now.

File: art/utils.py (hunk 1 above)
Comment (abigailgold) [reply art/utils.py]: Done.

**Comment by abigailgold** [comment]
@beat-buesser Thanks for the review, I added a unit test and fixed the formatting issue.

**Review by beat-buesser — APPROVED** [review]
Hi @abigailgold Thank you very much!
